### PR TITLE
Added command line parsing and first command

### PR DESCRIPTION
**Pull request by undercase**
This pull request sets up a basic structure for parsing commands using `sys.argv`, and features a command to generate a sqlite3 database for diary logs. The database is generated at a default path of `log.sqlite3`, so I changed DiaryDB to use that as the default path for the database file if no path is provided.

### CODE REVIEW

File: bin/diary [hunk 1]
```diff
@@ -1,7 +1,58 @@
...
+    else:
+        path = 'log.sqlite3'
+    cwd = os.getcwd()
+    connection = sqlite3.connect(os.path.join(cwd, path))
+    cursor = connection.cursor()
+    cursor.execute('''
+    CREATE TABLE IF NOT EXISTS logs
+    (inputDT TIMESTAMP, level TEXT, log TEXT)
+    ''')
+    print('sqlite3 database generated at %s' % os.path.join(cwd, 'log.sqlite3'))
```
Comment (GreenVars) [inline bin/diary]: should say path here

**Comment by GreenVars** [comment]
What's the best way to test these scripts? Shouldn't be too hard to just use os.system (or a better stdlib way) and check the output but how can we get diary easily available on the command line?

File: bin/diary (hunk 1 above)
Comment (undercase) [inline bin/diary]: Am fixing now, good catch.

**Comment by undercase** [comment]
We can test the script by using os.system to run the python script with our desired input. That way, the script doesn't need to be installed on the command line. It can be installed locally and tested during development by running `python setup.py install`.

File: bin/diary (hunk 1 above)
Comment (undercase) [inline bin/diary]: This has been fixed in latest commit.

**Comment by undercase** [comment]
I just added testing to the commands.

File: bin/diary [hunk 2]
```diff
@@ -1,7 +1,58 @@
...
+    if '-h' in sys.argv or '--help' in sys.argv:
+        help()
+    elif 'generate' in sys.argv:
+        generate(sys.argv)
+    else:
+        print("Diary couldn't parse that command.")
+        help()
+
+# Help command
+def help():
```
Comment (GreenVars) [inline bin/diary]: This shadows a builtin name, could you name this diary_help or something along those lines.

**Comment by GreenVars** [comment]
Could you write a run_tests.py in bin/diary/tests similar to the one in diary/tests? Add it the .travis.yml if you can

File: bin/diary (hunk 2 above)
Comment (undercase) [inline bin/diary]: Yes, I can. Will fix.

**Comment by undercase** [comment]
I've added run_tests.py for the command line tests and added it to the travis.yml. I have also changed the name of the `help()` function to `diary_help()`.

**Comment by GreenVars** [comment]
Subprocess should be the goto for command line in testing. It should also make testing easier as you can easily get access to output. https://docs.python.org/3/library/subprocess.html

**Comment by undercase** [comment]
I've gone down the subprocess and the os.system path (check out undercase/DarkPython), for this case os.system will definitely be easier to use. Subprocess is tough to use and gives us little benefit at the moment, since generators can be tested by just verifying the existence of files.

**Comment by undercase** [comment]
Getting subprocess to work locally and on Travis CI will probably be hell. It is better for most use cases, but it usually doesn't work for running other python programs within a python program because of weird issues with the GIL.

**Comment by GreenVars** [comment]
Would it be possible to have the test just import the diary scripts file and call the diary function with select arguments? Each test could have a line with command = "diary -h etc" and argparse could be used to feed those arguments into the command line function. https://docs.python.org/2/library/argparse.html#module-argparse

**Comment by undercase** [comment]
I don't think it's possible to run certain sections of code with arbitrary argument values. I have written a testing suite that is perfectly functional for developers, but I don't think Travis CI's security features will allow efficient testing of command line tools. I think we should merge this pull request and then I'll create an Issue for further research into this.

**Comment by GreenVars** [comment]
I got the tests working and implemented, can't figure out how to get the coverage to pick it up but that's not a big deal.

**Pull request merged**